nixos/modules/programs/opengamepadui.nix

```diff
@@ -0,0 +1,295 @@
+      mkdir -p "$(dirname "$GAMESCOPE_MODE_SAVE_FILE")"
+      touch "$GAMESCOPE_MODE_SAVE_FILE"
+
+      # Make path to Gamescope edid patched file.
+      mkdir -p "$(dirname "$GAMESCOPE_PATCHED_EDID_FILE")"
+      touch "$GAMESCOPE_PATCHED_EDID_FILE"
+
+      # Initially write no_display to our config file
+      # so we don't get mangoapp showing up before OpenGamepadUI initializes
+      # on OOBE and stuff.
+      mkdir -p "$(dirname "$MANGOHUD_CONFIGFILE")"
+      echo "no_display" >"$MANGOHUD_CONFIGFILE"
+
+      # Prepare our initial VRS config file
+      # for dynamic VRS in Mesa.
+      mkdir -p "$(dirname "$RADV_FORCE_VRS_CONFIG_FILE")"
+      echo "1x1" >"$RADV_FORCE_VRS_CONFIG_FILE"
+
+      # To play nice with the short term callback-based limiter for now
+      export GAMESCOPE_LIMITER_FILE=$(mktemp /tmp/gamescope-limiter.XXXXXXXX)
+
+      ulimit -n 524288
+
+      # Setup socket for gamescope
+      # Create run directory file for startup and stats sockets
+      tmpdir="$([[ -n ''${XDG_RUNTIME_DIR+x} ]] && mktemp -p "$XDG_RUNTIME_DIR" -d -t gamescope.XXXXXXX)"
+      socket="''${tmpdir:+$tmpdir/startup.socket}"
+      stats="''${tmpdir:+$tmpdir/stats.pipe}"
+
+      # Fail early if we don't have a proper runtime directory setup
+      if [[ -z $tmpdir || -z ''${XDG_RUNTIME_DIR+x} ]]; then
+      	echo >&2 "!! Failed to find run directory in which to create stats session sockets (is \$XDG_RUNTIME_DIR set?)"
+      	exit 0
+      fi
+
+      export GAMESCOPE_STATS="$stats"
+      mkfifo -- "$stats"
+      mkfifo -- "$socket"
+
+      # Start gamescope compositor, log it's output and background it
+      echo gamescope ${builtins.toString cfg.gamescopeSession.args}  -R $socket -T $stats >"$HOME"/.gamescope-cmd.log
+      gamescope ${builtins.toString cfg.gamescopeSession.args} -R $socket -T $stats >"$HOME"/.gamescope-stdout.log 2>&1 &
+      gamescope_pid="$!"
+
+      if read -r -t 3 response_x_display response_wl_display <>"$socket"; then
+      	export DISPLAY="$response_x_display"
+      	export GAMESCOPE_WAYLAND_DISPLAY="$response_wl_display"
+      	# We're done!
+      else
+      	echo "gamescope failed"
+      	kill -9 "$gamescope_pid"
+      	wait -n "$gamescope_pid"
+      	exit 1
+      	# Systemd or Session manager will have to restart session
+      fi
+
+      # If we have mangoapp binary start it
+      if command -v mangoapp >/dev/null; then
+      	(while true; do
+          sleep 1
+      		mangoapp >"$HOME"/.mangoapp-stdout.log 2>&1
+      	done) &
+      fi
+
+      # Start OpenGamepadUI
+      opengamepadui ${builtins.toString cfg.args}
+
+      # When the client exits, kill gamescope nicely
+      kill $gamescope_pid
+    '';
+
+  gamescopeSessionFile =
+    (pkgs.writeTextDir "share/wayland-sessions/opengamepadui.desktop" ''
+      [Desktop Entry]
+      Name=opengamepadui
+      Comment=OpenGamepadUI Session
+      Exec=${opengamepadui-gamescope}/bin/opengamepadui-gamescope
+      Type=Application
```

Comment:
Why not `makeDesktopItem`?

Reply:
Is there a way I can use `makeDesktopItem` to write the `.desktop` file to the `/share/wayland-sessions` folder? I was looking at the source for this and it seems like it only writes the file to `/share/applications`.

Reply:
Nope, but you should be able to create a symlink if needed.

Reply:
If the Wayland session file is saved to `/share/applications`, then it will show up as an application that can be launched from the desktop, which is not something we want. This `.desktop` file should only be available as a Wayland session.

Reply:
I see what the issue is now. I'll put together a PR to add that functionality to `makeDesktopItem`. For the time being, what you have is fine.